common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -82,3 +83,29 @@ def get_permission_object(self):
         can access these endpoints, we have to sort out the operator role use case
         """
         return self.requested_enterprise_uuid
+
+    def create(self, request, *args, **kwargs):
+        """
+        Process POST /auth/saml/v0/provider_config/ {postData}
+        """
+
+        # Create the samlproviderconfig model first
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        # Associate the enterprise customer with the provider
+        customer_uuid = self.requested_enterprise_uuid
+        try:
+            enterprise_customer = EnterpriseCustomer.objects.get(pk=customer_uuid)
+        except EnterpriseCustomer.DoesNotExist:
+            raise ValidationError('Enterprise customer not found at uuid: {}'.format(customer_uuid))
```

Comment:
Hmm, do you want to create the `SamlProviderConfig` instance above if you can't find an `EnterpriseCustomer` to go with it?  Maybe it would be better to do `self.perform_create(serializer)` as the second line of your `try` block.

Reply:
aha! good point yes I think we should only create this once enterprise customer has been found, will fix it thanks ... esp if there is no foreign key constraint it may 'pass' silently, and even if it fails constraint a 500 won't look good